internal/cmd/upstream/cmd.go

```diff
@@ -0,0 +1,104 @@
+package new
+
+import (
+	"io/ioutil"
+	"log"
+	"net/url"
+	"path/filepath"
+
+	"github.com/chirino/graphql-gw/internal/cmd/root"
+	"github.com/chirino/graphql-gw/internal/cmd/serve"
+	"github.com/chirino/graphql-gw/internal/gateway"
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+)
+
+var (
+	Command = &cobra.Command{
+		Use:   "add-upstream",
+		Short: "Adds new upstream to config.",
+		Long:  `Command lets you assemble gateway config by letting you add new upstream gateway`,
+		Run:   run,
+		Args:  cobra.MinimumNArgs(1),
+	}
+	UpstreamName = ""
+	UpstreamURL  = ""
+	ConfigFile   = ""
+)
+
+func init() {
+	Command.Flags().StringVar(&ConfigFile, "config", "graphql-gw.yaml", "path to the config file to load")
+	Command.Flags().StringVar(&UpstreamName, "name", "", "name of the upstream")
+	Command.Flags().StringVar(&UpstreamURL, "url", "", "url to the upstream")
+	root.Command.AddCommand(Command)
+}
+func run(cmd *cobra.Command, args []string) {
+	vebosityFmt := "%v"
+	if !root.Verbose {
+		vebosityFmt = "%+v\n"
+	}
+
+	file, err := ioutil.ReadFile(ConfigFile)
+
+	if err != nil {
+		log.Fatalf("Error reading config file: %s.", err)
+		panic(err)
+	}
+
+	if len(UpstreamName) > 0 {
+		log.Fatalf("Name parameter is missing")
+		return
+	}
+
+	if len(UpstreamURL) > 0 {
+		log.Fatalf("Name parameter is missing")
+		return
+	}
+
+	url, err := url.Parse(UpstreamURL)
+	if err != nil {
+		log.Fatalf(vebosityFmt, err)
+		panic(err)
+	}
+
+	log.Printf(`Contacting upstream %s`, url)
+
+	// TODO refactor loadEndpointSchema
```

Comment:
So the idea here is that I will look for Query, Mutation and Subscription types in the remote schema and mount them here